### ISSUE TITLE
FastText for semi supervised text classification

### ISSUE THREAD

**Issue by nguyenhoan1988** [facebookresearch/fastText#103]
I have a big dataset of Tweets (>100M) and only a small portion of those are labelled (~10K).
I would like to use fasttext to train a model to detect the topic of a tweet.
Do you have any methods to get use of all the data?
One way I could think of is train the word embedding model on the unlabelled tweets, then train the classifier on the labelled one.

**Comment by EdouardGrave** [comment]
Hi @nguyenhoan1988,
As of now, a potential solution is to learn unsupervised word vectors on the unlabeled tweets (e.g. using `./fasttext skipgram`) and then use these vectors to initialize the classifier, using the `-pretrainedVectors` command line option. This option is used to specify a text file containing pre-trained word vectors (e.g. the `.vec` file outputed by `./fasttext skipgram`).

```
./fasttext skipgram -input UNLABELED_DATA -output UNSUP_VECTORS
./fasttext supervised -input LABELED_DATA -output MODEL -pretrainedVectors UNSUP_VECTORS.vec
```

**Comment by ebuildy** [comment]
The main problem with this approach is fasttext will predict labels for any thing. I found much better to have a kind of label ``common`` , to put all the common sentence/terms.
